File: backend/myPizza/src/main/java/backendapp/myPizza/controllers/SessionController.java
package backendapp.myPizza.controllers;

import backendapp.myPizza.Models.entities.WorkSession;
import backendapp.myPizza.Models.reqDTO.StartSessionDTO;
import backendapp.myPizza.Models.resDTO.ConfirmRes;
import backendapp.myPizza.Models.resDTO.DeliveryTimeRes;
import backendapp.myPizza.Models.resDTO.IsThereAnActiveSessionRes;
import backendapp.myPizza.Models.resDTO.TimeIntervalsRes;
import backendapp.myPizza.exceptions.BadRequestException;
import backendapp.myPizza.exceptions.Validation;
import backendapp.myPizza.services._SessionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.validation.BindingResult;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/work-session")

public class SessionController {

    @Autowired
    _SessionService _sessionSvc;

    @PreAuthorize("hasAuthority('ADMIN')")
    @GetMapping("/get-active-session")
    public WorkSession getActiveSession() throws BadRequestException {
        return _sessionSvc.getActiveSession();
    }

    @GetMapping("/is-there-an-active-session")
    public IsThereAnActiveSessionRes isThereAnActiveSession() {
        return new IsThereAnActiveSessionRes(_sessionSvc.isThereAnActiveSession());
    }

    @PreAuthorize("hasAuthority('ADMIN')")
    @PostMapping("/start-new-session")
    public WorkSession startNewSession(@RequestBody @Validated StartSessionDTO startSessionDTO, BindingResult validation) throws BadRequestException {
        Validation.validate(validation);
        return _sessionSvc.startNewSession(startSessionDTO);
    }

    @GetMapping("/get-delivery-times")
    public DeliveryTimeRes getDeliveryTimes() {
        return new DeliveryTimeRes(_sessionSvc.getActiveSessionDeliveryTimes());
    }
    @PreAuthorize("hasAuthority('ADMIN')")
    @GetMapping("/get-active-session-time-intervals")
    public TimeIntervalsRes getActiveSessionTimeIntervals() {
        return new TimeIntervalsRes(_sessionSvc.getActiveSessionTimeIntervals());
    }

    @PreAuthorize("hasAuthority('ADMIN')")
    @GetMapping("/close")
    public ConfirmRes closeActiveSession() throws BadRequestException {
        return _sessionSvc.closeActiveSession();
    }




}

File: backend_nodejs_nest_fastify/my-pizza/src/nest_modules/order/controllers/session.controller.ts
import { IsThereAnActiveSessionRes } from './../interfaces/is-there-an-active-session-res.interface';
import { Body, Controller, Get, Post, UseGuards } from '@nestjs/common';
import { AdminGuard } from 'src/nest_modules/auth-user/guards/admin.guard';
import { SessionService } from '../services/session.service';
import { WorkSession } from '../entities/work-session.entity';
import { StartSessionDTO } from '../interfaces/start-session-dto.interface';
import { DeliveryTimesRes } from '../interfaces/delivery-times-res.interface';
import { TimeIntervalsRes } from '../interfaces/time-intervals-res.interface';
import { ConfirmRes } from 'src/nest_modules/auth-user/interfaces/confirm-res.interface';

@Controller('api/work-session')
export class SessionController {

    constructor(private _session: SessionService) { }

    @UseGuards(AdminGuard)
    @Get('/get-active-session')
    public async getActiveSession(): Promise<WorkSession> {
        return await this._session.getActiveSession()
    }

    @Get('/is-there-an-active-session')
    public async isThereAnActiveSession(): Promise<IsThereAnActiveSessionRes> {
        return {
            isThereAnActiveSession: await this._session.isThereAnActiveSession()
        }
    }

    @Post('/start-new-session')
    public async startNewSession(@Body() startSessionDTO: StartSessionDTO): Promise<WorkSession> {
        return await this._session.startNewSession(startSessionDTO)
    }

    @Get('/get-delivery-times')
    public async getDeliveryTimes(): Promise<DeliveryTimesRes> {
        return {
            deliveryTimes: await this._session.getPublicSessionDeliveryTimes()
        }
    }

    @UseGuards(AdminGuard)
    @Get('/get-active-session-time-intervals')
    public async getActiveSessionTimeIntervals(): Promise<TimeIntervalsRes> {
        return {
            timeIntervals: await this._session.getActiveSessionTimeIntervals()
        }
    }

    @UseGuards(AdminGuard)
    @Get('/close')
    public async closeActiveSession(): Promise<ConfirmRes> {
        return await this._session.closeActiveSession()
    }


}
